moveit_core/robot_model/include/moveit/robot_model/joint_model_group.h

```diff
@@ -166,6 +166,19 @@ class JointModelGroup
     return active_joint_model_name_vector_;
   }
 
+  /** \brief Get the names of the active variables that make up the joints included in this group. The number of
+      returned elements is always equal to getVariableCount(). This does not include mimic joints. */
+  const std::vector<std::string>& getActiveVariableNames() const
```

Comment:
What's wrong with getActiveJointModelNames() https://moveit.picknik.ai/humble/api/html/classmoveit_1_1core_1_1JointModelGroup.html#ad3a2b2661f3303f1d6fb8a8a08008174

Reply:
... ok I see. The fact that you have to do something like this really shows that the API and mapping implementation of RobotState and joints is very awkward. We know that TOTG only supports active joints with single variables, and in that case the variable names always equal the joint model names (https://moveit.picknik.ai/humble/api/html/classmoveit_1_1core_1_1JointModel.html#a09b0ad5f0d57c67a1fd7cde8d494519d). But that's convention and not clear from the API. I'm not against adding these functions, but they somewhat make the API even more complicated. Would it help to just expose and use this function instead https://moveit.picknik.ai/humble/api/html/classmoveit_1_1core_1_1JointModelGroup.html#aa11b464a43bb475addb128a9f1b8bbb1?

Reply:
I also agree that adding more "getters" is awkward, so if the suggested function makes this a little better, I'm open to try it.

I also have to find time to figure out how/why this breaks rolling but not humble, as the changes are purely additive from what I can tell.

Reply:
maybe related to this? https://github.com/ros-planning/moveit2/pull/1821

Reply:
Looking through, I think the test is just fundamentally broken, actually.

The test initializes a trajectory of 5 waypoints in [this function](https://github.com/ros-planning/moveit2/blob/main/moveit_core/robot_trajectory/test/test_robot_trajectory.cpp#L68-L89), but all the waypoints have the same exact robot state... so it's actually correct that the path length is 0.0. But the test wants it to be (greater than 0](https://github.com/ros-planning/moveit2/blob/70af62818f0e0be8c572c2f51b3357854418b5fd/moveit_core/robot_trajectory/test/test_robot_trajectory.cpp#L326).